app/address/[address]/layout.tsx

```diff
@@ -275,20 +280,32 @@ function AccountHeader({
         let token: { logoURI?: string; name?: string } = {};
         let unverified = false;
 
-        const metadataExtension = mintInfo?.extensions?.find(({ extension }: { extension: string }) => extension === 'tokenMetadata');
-        const metadataPointerExtension = mintInfo?.extensions?.find(({ extension }: { extension: string }) => extension === 'metadataPointer');
-
+        const metadataExtension = mintInfo?.extensions?.find(
+            ({ extension }: { extension: string }) => extension === 'tokenMetadata'
+        );
+        const metadataPointerExtension = mintInfo?.extensions?.find(
+            ({ extension }: { extension: string }) => extension === 'metadataPointer'
+        );
 
         if (metadataPointerExtension && metadataExtension) {
             const tokenMetadata = create(metadataExtension.state, TokenMetadata);
```

Comment:
I tried to implement your code suggestion but unfortunately didn't succeed. It seems that `tokenMetadata` is available only within this conditional, which is why I added the IIFE function within `AccountHeader`.

From my testing, I haven't noticed any inconsistencies with the state, but please let me know if you observe otherwise.

Reply:
hmm okay, gotcha. i think this is okay for now

can you do some validation on the response / some error handling?

Reply:
@jnwng I just added a basic error handling block.